### PR TITLE
feat: updates for transfers from vested accounts

### LINKED ISSUE #2032: Transfers from a vested account should
Transfers from a vested account to general account of the same key should
- incur no transfer fee
- not be subject to minimum transfer amount if the full amount is transferred

### PR DESCRIPTION
**Pull request by cdummett**
closes #2032

### CODE REVIEW

File: protocol/0057-TRAN-transfers.md [hunk 1]
```diff
@@ -153,7 +155,7 @@ Note: if there is no market with contribution to the reward metric - no transfer
 
 ## Fees
 
-A fee is taken from all transfers, and paid out to validators in a similar manner to the existing [infrastructure fees](0061-REWP-pos_rewards.md). For recurring transfers, the fee is charged each time the transfer occurs.
+A fee is taken from all transfers (except transfers from a vested account), and paid out to validators in a similar manner to the existing [infrastructure fees](0061-REWP-pos_rewards.md). For recurring transfers, the fee is charged each time the transfer occurs.
```
Comment (davidsiska-vega) [inline protocol/0057-TRAN-transfers.md]:
```suggestion
A fee is taken from all transfers (except transfers from a vested account to a general account held by the same key), and paid out to validators in a similar manner to the existing [infrastructure fees](0061-REWP-pos_rewards.md). For recurring transfers, the fee is charged each time the transfer occurs.
```

File: protocol/0057-TRAN-transfers.md [hunk 2]
```diff
@@ -243,6 +245,9 @@ message CancelTransfer {
   - If I have enough funds to pay transfer and fees, the transfer happens.
   - If I do not have enough funds to pay transfer and fees, the transfer is cancelled.
   - The fees are being paid into the infrastructure pool
+- As a user I can do a transfer from a vested account without incurring any fees (<a name="0057-TRAN-066" href="#0057-TRAN-066">0057-TRAN-066</a>).
```
Comment (davidsiska-vega) [inline protocol/0057-TRAN-transfers.md]:
```suggestion
- As a user I can do a transfer from a vested account to a general account held by the same key without incurring any fees (<a name="0057-TRAN-066" href="#0057-TRAN-066">0057-TRAN-066</a>).
- As a user I can do a transfer from a vested account to another account that isn't general account for my key *but* in this case transfer fees are incurred (<a name="0057-TRAN-069" href="#0057-TRAN-069">0057-TRAN-069</a>).
```

File: protocol/0085-RVST-rewards_vesting.md [hunk 3]
```diff
@@ -32,7 +32,7 @@ When transferring funds from the vesting account to the vested account, a new tr
 
 ## Rewards bonus multiplier
 
-Once vested rewards are transferred to the vested account, the party will be able to transfer funds to their general account using a normal transfer.
+Once vested rewards are transferred to the vested account, the party will be able to transfer funds to their general account using a normal transfer. When transferring funds from a vested account a party will incur no transfer fees and if transferring the full balance they will not be subject to the minimum quantum transfer amount.
```
Comment (davidsiska-vega) [inline protocol/0085-RVST-rewards_vesting.md]:
```suggestion
Once vested rewards are transferred to the vested account, the party will be able to transfer funds to their general account using a normal transfer.
When transferring funds from a vested account to the general account held by their key a party will incur no transfer fees and if transferring the full balance they will not be subject to the minimum quantum transfer amount.
```

File: protocol/features.json [hunk 4]
```diff
@@ -112,6 +112,14 @@
       "0057-TRAN-065"
     ]
   },
+  "Transfers from vesting": {
+    "milestone": "palazzo",
+    "acs": [
+      "0057-TRAN-066",
+      "0057-TRAN-067",
+      "0057-TRAN-068"
```
Comment (davidsiska-vega) [inline protocol/features.json]: "0057-TRAN-069",

**Review by davidsiska-vega — CHANGES_REQUESTED** [review]
I've left a few suggestions, please have a look.

**Review by davidsiska-vega — DISMISSED** [review]

File: protocol/0057-TRAN-transfers.md [hunk 5]
```diff
@@ -39,9 +39,11 @@ In order to prevent the abuse of user-initiated transfers as spam attack there w
 
 ## Minimum transfer amount
 
-This is controlled by the `transfer.minTransferQuantumMultiple` and quantum specified for the [asset](0040-ASSF-asset_framework.md)).
+This is controlled by the `transfer.minTransferQuantumMultiple` and quantum specified for the [asset](0040-ASSF-asset_framework.md).
 The minimum transfer amount is `transfer.minTransferQuantumMultiple x quantum`.
 
+If a user is transferring funds from a vested account, if their balance (expressed in quantum) is less than the minimum amount, they should be able to transfer the full balance (note, transferring less then the full balance is not permitted).
```
Comment (TomMcL) [inline protocol/0057-TRAN-transfers.md]:
```suggestion
If a user is transferring funds from a vested account, if their balance (expressed in quantum) is less than the minimum amount, they should be able to transfer the full balance (note, transferring less than the full balance is not permitted).
```

**Review by davidsiska-vega — APPROVED** [review]

**Comment by gordsport** [comment]
Core change in:

- https://github.com/vegaprotocol/vega/pull/9984

**Pull request merged**